src/Controller/AdminResettingController.php

```diff
@@ -77,7 +76,11 @@ public function sendEmailAction(Request $request)
                 $user->setConfirmationToken($tokenGenerator->generateToken());
             }
 
-            $this->sendResettingEmailMessage($user);
+            $mailerService = $this->getParameter('sonata.user.mailer.service');
+            /** @var MailerInterface $mailer */
+            $mailer = $this->get($mailerService);
```

Comment:
don't we want to get away from `$this->get`?

Reply:
Yes, if this is the (good!) move what is the preferred way to change it in sonata?

Reply:
You would have to make the `AdminResettingController` a controller as service (this is what we are trying to do, see for instance https://github.com/sonata-project/SonataAdminBundle/pull/5120), get its definition in the extension of the user bundle, and then modify it to pass the mailer service as an argument. It might be a lot of work though, maybe we can skip it for now? If you feel like making this change, please create a PR dedicated to making this controller a CaS first.

Reply:
Is there some documentation about CaaSes actions?

Reply:
Here you go: https://symfony.com/doc/current/controller/service.html
We are also moving to invokable CaSes https://symfony.com/doc/current/controller/service.html#invokable-controllers  (same page, particular paragraph)

Reply:
Ok I found these resources yesterday and as the pull request on admin bundle was discussed a lot I think I got the big picture. So let's go :) I will open another PR soon about it!